modules/ROOT/pages/creating-an-odatav4-api-with-apikit.adoc

```diff
@@ -93,22 +93,125 @@ image::odata-query-option-attributes.png["odataRequestAttributes is highlighted
 
 == Use The Transform to SQL Operation
 
-APIkit for OData 4 contains a `Transform to SQL Select` operation to transform the input OData `GET` request to a SQL SELECT query. The operation is useful when combined with the `select` operation of the Database Connector. The output payload is a prepared statement and the attributes are a map of prepared statement parameters.
+APIkit for OData 4 contains a `Transform to SQL Select` operation to transform the input OData `GET` request to a SQL SELECT query. The operation is useful when combined with the `select` operation of the Database Connector. The output payload is a prepared statement and the attributes are a map of prepared statement parameters. It has been designed to work with the Mule DB connector to help the user expose databases with OData 4 easily.
 
 image::transform-to-sql-transformer.png["Transform to SQL select element and tab are highlighted."]
 
 image::add-db-transform-to-sql.png["Select element and tab are highlighted, as well as the attributes.queryparameters in the input parameters field."]
 
 image::serialize-transform-to-sql.png["Serialized entity element and tab are highlighted, as well as the inbound content in the General section."]
 
+On the one hand, APIkit currently supports the following elements:
+
+* System query options:
+** $filter
+** $select
+** $search
+** $orderBy
+** $skip
+** $top
+* Single entity
+* Entity relation
+* Server-side pagination
+
+On the other hand, APIkit does not currently support the following elements:
+
+* Parameter Aliases
+* Built-in Filter Operations:
+** has
+** divby 
+* Built-in Query Functions, all but:
+** contains
+** startswith
+** endwith
+* $apply
+* $compute
+* $expand
+
+=== How does transform to SQL select work in OData 4?
+
+The OData protocol defines a series of conventions that allow clients to query identified resources using URLs. The list of queryable entities, their fields, and the relationships between them are defined in a data model using simple HTTP messages.
+As OData shares some similarities with Java database connectivity (JDBC), the `transform-to-sql-select` operation takes advantage of such similarities to help the user create SQL queries for those resources. Then, the query is returned at the payload along with the query parameters at the attributes.
```

Comment:
"Then, the query is returned at the payload along with the query parameters at the attributes."
This sentence is unclear, please address the following points: 
-"the query is returned" > who or what returns the query? Avoid passive voice by specifying **what** returns the query
-"returned at the payload" > I can't understand this, does this mean "returned **to** the payload?"
-"query parameters at the attributes" > also confusing, was this intended to be "**as** the attributes" or "**and** the attributes" ? 
Please check with devs to clarify these points.

Reply:
`transform-to-sql` is an operation that takes a description of a given API request as input and outputs a SQL query (as payload) and a "parameter -> value" mapping (as attributes).

Reply:
Hi, APIkit team!
Please, can you further explain this:
"query parameters at the attributes".
Is this intended to be "as the attributes" or "and the attributes"?
Apart from that, please, can you take a look at line 143 ("As OData shares... with the query parameters at the attributes"). Is the current explanation accurate?
Thanks for your support!

Reply:
The explanation is accurate but the last sentence sounds disconnected and out of context, it might be rephrased. As @iglosiggio mentioned, the idea is to explain the structure of the resulting Mule message of the operation (see [mule message documentation](https://docs.mulesoft.com/mule-runtime/4.4/about-mule-message) as reference)

Reply:
The resulting Mule message ends up being a payload with a prepared statement of the SQL `SELECT` sentence as plain text, along with a map of attributes where keys are the name of the variables in the prepared statement and values the actual values of each variable.

Reply:
If you look at the output of the operation you will see that the resulting Mule Message has a modified `Payload` and `Attributes` (typed accordingly... well, I think `Payload` should be typed as an `String` but nvm).
![image](https://user-images.githubusercontent.com/1154680/170129210-a66c951c-dfe4-4e26-b666-f204199a5441.png)

I don't know what's the best wording to express this. But the basic idea is that the payload of the Mule Message outputted by this operation _is_ the query and the attributes _is_ the query parameters mapping.

Does this help?

Reply:
Resolved over chat! ;)